File: benchmark/c/cpw/nth_non_fibonacci_number/nth_non_fibonacci_number.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int nth_non_fibonacci_number ( int n ) ;
int nth_non_fibonacci_number ( int n ) {
  int prevPrev = 1, prev = 2, curr = 3;
  while ( n > 0 ) {
    prevPrev = prev;
    prev = curr;
    curr = prevPrev + prev;
    n = n - ( curr - prev - 1 );
  }
  n = n + ( curr - prev - 1 );
  return prev + n;
}

File: translation_result/else_bms/fail/c/cpw/nth_non_fibonacci_number/gemini/Restart/nth_non_fibonacci_number.rs

fn nth_non_fibonacci_number(mut n: i32) -> i32 {
    let mut prev_prev = 1;
    let mut prev = 2;
    let mut curr = 3;
    while n > 0 {
        prev_prev = prev;
        prev = curr;
        curr = prev_prev + prev;
        n -= (curr - prev - 1);
    }
    n += (curr - prev - 1);
    prev + n
}
